### DISCUSSION TITLE
Error 403 when git tag

### DISCUSSION THREAD

**Discussion by mregnauld** [codemagic-ci-cd/codemagic-docs#1398] [Q&A] [answered]
In my workflow, I try to add a tag to my repo by doing the following:

```shell
git tag -a v1.5.4 -m "Version 1.5.4"
git push -q origin :refs/tags/v1.5.4
git push origin refs/tags/v1.5.4
```

But then I get the following error:

```
remote: Write access to repository not granted.
fatal: unable to access 'https://github.com/mregnauld/mistikee/': The requested URL returned error: 403
remote: Write access to repository not granted.
fatal: unable to access 'https://github.com/mregnauld/mistikee/': The requested URL returned error: 403
```

Unfortunately, I can't find any way to add the needed authorization.

How can I fix that?

Thanks.

**Comment by icarusdust** [comment, marked as answer]
Hi @mregnauld, check out [this documentation](https://docs.codemagic.io/knowledge-git/add-a-git-tag-with-app-version/) where creating a tag with app version and pushing it explained.

**Reply by mregnauld** [reply]
It works now, thanks!

That being said, I think that there is a little mistake in the documentation (or at least, it's a bit confusing).

The following line wasn't working for me:
```shell
git push "https://your-username:$APP_PASSWORD@your-git-service.com/your-repo.git" --tags
```

I had to change it to:
```shell
git push "https://your-username:$APP_PASSWORD@your-git-service.com/your-username/your-repo.git" --tags
```

But maybe it's specific to Github.
